File: src/depsolver/python/model/command.py
class Install(object):
    def __init__(self, constraint):
        self.constraint = constraint

    def is_install(self):
        return True

    def __str__(self):
        return "+" + str(self.constraint)

    __repr__ = __str__

class Uninstall(object):
    def __init__(self, constraint):
        self.constraint = constraint

    def is_install(self):
        return False

    def __str__(self):
        return "-" + str(self.constraint)

    __repr__ = __str__

File: src/depsolver/rust/src/model/command.rs
use std::fmt;
use super::constraint::PackageConstraint;

enum Command<'a> {
   Install(PackageConstraint<'a>),
   Uninstall(PackageConstraint<'a>)
}

impl<'a> fmt::Display for Command<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match &self {
            Command::Install(c) =>
                write!(f, "+{}", c),
            Command::Uninstall(c) =>
                write!(f, "-{}", c)
        }
    }
}
